src/Sylius/Bundle/ApiBundle/Controller/GetSalesStatisticsAction.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\ApiBundle\Query\Admin\GetSalesStatistics;
+use Sylius\Component\Core\Sales\ValueObject\SalesPeriod;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+use Symfony\Component\Serializer\SerializerInterface;
+use Webmozart\Assert\Assert;
+
+final class GetSalesStatisticsAction
+{
+    public function __construct(private MessageBusInterface $queryBus, private SerializerInterface $serializer)
+    {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $channelCode = $request->query->get('channelCode');
+
+        if ($channelCode === null) {
+            return new JsonResponse(['error' => 'Missing channelCode parameter.'], Response::HTTP_BAD_REQUEST);
+        }
+
+        $salesPeriod = new SalesPeriod(
```

Comment:
For now, it's hardcoded, but eventually it will be fully customizable by the client